extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ClassConfigPropertiesUtil.java

```diff
@@ -0,0 +1,359 @@
+package io.quarkus.arc.deployment.configproperties;
+
+import static io.quarkus.arc.deployment.configproperties.ConfigPropertiesUtil.ReadOptionalResponse;
+import static io.quarkus.arc.deployment.configproperties.ConfigPropertiesUtil.createReadMandatoryValueAndConvertIfNeeded;
+import static io.quarkus.arc.deployment.configproperties.ConfigPropertiesUtil.createReadOptionalValueAndConvertIfNeeded;
+import static io.quarkus.arc.deployment.configproperties.ConfigPropertiesUtil.determineSingleGenericType;
+
+import java.lang.reflect.Modifier;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.Config;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.FieldInfo;
+import org.jboss.jandex.IndexView;
+import org.jboss.jandex.MethodInfo;
+import org.jboss.jandex.Type;
+
+import io.quarkus.deployment.bean.JavaBeanUtil;
+import io.quarkus.deployment.util.HashUtil;
+import io.quarkus.gizmo.BranchResult;
+import io.quarkus.gizmo.BytecodeCreator;
+import io.quarkus.gizmo.ClassCreator;
+import io.quarkus.gizmo.ClassOutput;
+import io.quarkus.gizmo.FieldCreator;
+import io.quarkus.gizmo.FieldDescriptor;
+import io.quarkus.gizmo.MethodCreator;
+import io.quarkus.gizmo.MethodDescriptor;
+import io.quarkus.gizmo.ResultHandle;
+import io.quarkus.runtime.StartupEvent;
+
+final class ClassConfigPropertiesUtil {
+
+    private static final String JAVAX_VALIDATION_PREFIX = "javax.validation";
+    private static final String VALIDATOR_CLASS = "javax.validation.Validator";
+    private static final String PATH = "javax.validation.Path";
+    private static final String CONSTRAINT_VIOLATION_CLASS = "javax.validation.ConstraintViolation";
+    private static final String HIBERNATE_VALIDATOR_IMPL_CLASS = "org.hibernate.validator.internal.engine.ValidatorImpl";
+    private static final String CONSTRAINT_VIOLATION_EXCEPTION_CLASS = "javax.validation.ConstraintViolationException";
+
+    private ClassConfigPropertiesUtil() {
+    }
+
+    /**
+     * Generates a class like the following:
+     *
+     * <pre>
+     * &#64;ApplicationScoped
+     * public class EnsureValidation {
+     *
+     *     &#64;Inject
+     *     MyConfig myConfig;
+     *
+     *     &#64;Inject
+     *     OtherProperties other;
+     *
+     *     public void onStartup(@Observes StartupEvent ev) {
```

Comment:
Why do you need that if it's empty?

Reply:
The whole trick here is to force the beans to be produced eagerly. By injecting the config classes into this bean and adding an `@Observes StartupEvent` we ensure that Arc runs the producer thus ensuring validation happens at startup

Reply:
Wouldn't using `@Singleton` beans have the same consequence and avoid some boilerplate? I might be wrong, better check with @mkouba

Reply:
Instead of @ApplicationScoped you mean? That sounds reasonable to me.

I just used whatever I found in some existing code.

@mkouba should I change to `@Singleton`?

Reply:
From what I understood, singletons are always initialized eagerly. So that might solve your issue and you might be able to drop the `onStartup` method. But I might be wrong on that :).

Reply:
We discussed with @mkouba and the current implementation is the probably the best way to guarantee it always works

Reply:
> From what I understood, singletons are always initialized eagerly.

Not really. A singleton is initialized eagerly when injected. This bean is used to initialize other beans when the app starts. That's why we use `StartupEvent` and inject other beans and invoke `toString()` upon them because a lazy proxy would be injected for a normal scoped bean.

Reply:
Ok, thanks for the explanation, Martin :).